File: app/src/main/java/com/application/foundation/features/common/model/RequestPermissionsModel.java
package com.application.foundation.features.common.model;

import androidx.annotation.NonNull;

public class RequestPermissionsModel extends RequestBase<Void, Void> {

	public static final String TAG = "RequestPermissionsModel";

	private int requestCode;
	private String[] permissions;
	private int[] grantResults;

	public void onRequestPermissionsResult(int requestCode,
										   @NonNull String[] permissions, @NonNull int[] grantResults) {
		this.requestCode = requestCode;
		this.permissions = permissions;
		this.grantResults = grantResults;
		notifyListeners();
	}

	public int getRequestCode() {
		return requestCode;
	}

	public String[] getPermissions() {
		return permissions;
	}

	public int[] getGrantResults() {
		return grantResults;
	}
}

File: app/src/main/java/com/application/foundation/features/common/model/RequestPermissionsModel.kt
package com.application.foundation.features.common.model

import kotlin.properties.Delegates

class RequestPermissionsModel : RequestBase<Any, Any>() {

	var requestCode: Int by Delegates.notNull()
		private set

	lateinit var permissions: Array<String>
		private set

	lateinit var grantResults: IntArray
		private set


	fun onRequestPermissionsResult(
		requestCode: Int,
		permissions: Array<String>, grantResults: IntArray
	) {
		this.requestCode = requestCode
		this.permissions = permissions
		this.grantResults = grantResults

		notifyListeners()
	}
}